### PR TITLE
Test news search

### PR DESCRIPTION
**Pull request by Francieli-S**
**!!! This test file depends on merging Regina's PR (search routes, validation, and controller) !!!**

This pull request will:

- add new file (search.test.ts) to  @./tests/news

- add validation test to the endpoint api/news/search:
1. Missing or invalid required q parameter returns a 400 error.
2. Optional parameters (from, to, sortBy) correctly.
3. Invalid date formats for from and to.
4. Invalid sortBy value.

- add test for API response:
1. Search articles by keyword q and sortBy
2. Filter articles within a date range (from and to).

image

### CODE REVIEW

File: src/tests/news/search.test.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import request from 'supertest';
+import app from '../../index';
+
+const getSearchResultsUrl = '/api/news/search';
+
+describe('GET /api/news/search', () => {
+  describe('Validation', () => {
+    it('should validate required q parameter and return a 400 if q is missing', async () => {
```
Comment (bojana-petroska) [inline src/tests/news/search.test.ts:8]: i would consider an empty line before each `it` for readability, as you have in some places

File: src/tests/news/search.test.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      const response = await request(app).get(getSearchResultsUrl);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBe('"q" is required');
+    });
+    it('should validate required q parameter and return a 400 if q is invalid', async () => {
```
Comment (bojana-petroska) [inline src/tests/news/search.test.ts:13]: i would consider an empty line before each `it` for readability, as you have in some places

File: src/tests/news/search.test.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        q: 'test',
+        sortBy: 'invalid: not relevancy, popularity, or publishedAt',
+      });
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBe(
+        '"sortBy" must be one of [relevancy, popularity, publishedAt]',
+      );
+    });
+  });
+  describe('API Response Tests', () => {
```
Comment (bojana-petroska) [inline src/tests/news/search.test.ts:57]: I like it that you separated the validation and API response tests!

Comment (Francieli-S) [reply src/tests/news/search.test.ts:57]: @bojana-petroska Thanks! I agree and will add the empty lines. I am using a Jest extension in VScode to run a single test case, and it adds an extra space right above the 'it' line, so in VScode it looks like an empty line!!!

**Comment by Francieli-S** [comment]
image

**Review by ahzia — APPROVED** [review]
Good Job @Francieli-S  👏👏👏
I am merging you code

**Pull request merged**